### PR TITLE
🌱 Remove previously deprecated code 

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:
follows https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md#track-remove-previously-deprecated-code

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of: https://github.com/kubernetes-sigs/cluster-api/issues/9094

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
Thanks!

**Comment by furkatgofurov7** [comment]
> Thanks!

I had to fix a typo in the markdown, and label was lost after a force push. PTAL once again, thank you.

File: cmd/clusterctl/client/cluster/cert_manager.go [hunk 1]
```diff
@@ -347,13 +341,9 @@ func (cm *certManagerClient) shouldUpgrade(objs []unstructured.Unstructured) (st
 		// if there is no version annotation, this means the obj is cert-manager v0.11.0 (installed with older version of clusterctl)
 		objVersion, ok := obj.GetAnnotations()[clusterctlv1.CertManagerVersionAnnotation]
 		if !ok {
-			// try the old annotation name
-			objVersion, ok = obj.GetAnnotations()[certManagerVersionAnnotation]
-			if !ok {
-				currentVersion = "v0.11.0"
-				needUpgrade = true
-				break
-			}
```
Comment (vincepri) [inline cmd/clusterctl/client/cluster/cert_manager.go:356]: The logic after this commit has changed though; now we're always defaulting to `v0.11.0` if the annotation isn't there; which might be ok. @fabriziopandini is this what we want to do?

File: cmd/clusterctl/client/cluster/cert_manager.go [hunk 2]
```diff
@@ -42,12 +42,6 @@ const (
 	waitCertManagerInterval = 1 * time.Second
 
 	certManagerNamespace = "cert-manager"
-
-	// This is maintained only for supporting upgrades from cluster created with clusterctl v1alpha3.
-	//
-	// Deprecated: Use clusterctlv1.CertManagerVersionAnnotation instead.
-	// TODO: Remove once upgrades from v1alpha3 are no longer supported.
-	certManagerVersionAnnotation = "certmanager.clusterctl.cluster.x-k8s.io/version"
```
[the remark this answers is not in the document]
Comment (killianmuldoon) [reply cmd/clusterctl/client/cluster/cert_manager.go:50]: I think unless we have specific migration for this annotation we can't remove it. Could you maybe take this piece out of the PR and open an issue to cover over migration tasks for this annotation?

[the remark this answers is not in the document]
Comment (furkatgofurov7) [reply cmd/clusterctl/client/cluster/cert_manager.go:50]: Issue created: https://github.com/kubernetes-sigs/cluster-api/issues/9145

Removing this particular piece out of this PR

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
Thx!

/lgtm

/assign @vincepri

**Review by Jont828 — COMMENTED** [review]
/lgtm

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**